api/grpcserver/v2alpha1/activation_test.go

```diff
@@ -227,9 +227,27 @@ func TestActivationService_ActivationsCount(t *testing.T) {
 	conn := dialGrpc(ctx, t, cfg)
 	client := spacemeshv2alpha1.NewActivationServiceClient(conn)
 
-	count, err := client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{
-		Epoch: activations[3].PublishEpoch.Uint32(),
+	t.Run("count without filter", func(t *testing.T) {
+		count, err := client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{})
+		require.NoError(t, err)
+		require.Len(t, activations, int(count.Count))
+	})
+
+	t.Run("count with filter", func(t *testing.T) {
+		epoch := uint32(1)
+		countOne, err := client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{
+			Epoch: &epoch,
+		})
+		require.NoError(t, err)
+		require.Greater(t, countOne.Count, uint32(0))
+
+		epoch = uint32(3)
+		countTwo, err := client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{
+			Epoch: &epoch,
+		})
+		require.NoError(t, err)
+		require.Greater(t, countTwo.Count, uint32(0))
+
+		require.NotEqual(t, countOne.Count, countTwo.Count)
```

Comment:
Written like this this test will occasionally randomly fail, when a) no ATXs for epoch 1 were generated b) no ATXs for epoch 3 were generated or c) epoch 1 and epoch 3 have the same number of ATXs.

Instead is suggest the following:

```go
// ...
	genEpoch3 := fixture.NewAtxsGenerator().WithEpochs(3, 1)
	epoch3ATXs := make([]types.VerifiedActivationTx, 30)
	for i := range epoch3ATXs {
		atx := gen.Next()
		require.NoError(t, atxs.Add(db, atx))
		activations[i] = *atx
	}
	genEpoch5 := fixture.NewAtxsGenerator().WithEpochs(5, 1)
	epoch5ATXs := make([]types.VerifiedActivationTx, 10) // ensure the number here is different from above
	for i := range epoch5ATXs {
		atx := gen.Next()
		require.NoError(t, atxs.Add(db, atx))
		activations[i] = *atx
	}
// ...
	epoch := uint32(3)
	count, err := client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{
		Epoch: &epoch,
	})
	require.NoError(t, err)
	require.Equal(t, len(epoch3ATXs), int(count.Count))

	epoch = uint32(5)
	count, err = client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{
		Epoch: &epoch,
	})
	require.NoError(t, err)
	require.Equal(t, len(epoch5ATXs), int(count.Count))

	count, err = client.ActivationsCount(ctx, &spacemeshv2alpha1.ActivationsCountRequest{})
	require.NoError(t, err)
	require.Equal(t, len(epoch3ATXs) + len(epoch5ATXs), int(count.Count))
```